web-bundle/src/main/java/com/graphhopper/resources/CustomWeightingRouteResource.java

```diff
@@ -0,0 +1,164 @@
+        long took = sw.stop().getNanos() / 1000;
+        String infoStr = httpReq.getRemoteAddr() + " " + httpReq.getLocale() + " " + httpReq.getHeader("User-Agent");
+        String queryString = httpReq.getQueryString() == null ? "" : (httpReq.getQueryString() + " ");
+        String logStr = queryString + infoStr + " " + request.getPoints().size() + ", took: "
+                + took + " micros, algo: " + request.getAlgorithm() + ", profile: " + request.getProfile()
+                + ", " + weightingVehicleLogStr;
+
+        if (ghResponse.hasErrors()) {
+            logger.error(logStr + ", errors:" + ghResponse.getErrors());
+            throw new MultiException(ghResponse.getErrors());
+        } else {
+            logger.info(logStr + ", alternatives: " + ghResponse.getAll().size()
+                    + ", distance0: " + ghResponse.getBest().getDistance()
+                    + ", weight0: " + ghResponse.getBest().getRouteWeight()
+                    + ", time0: " + Math.round(ghResponse.getBest().getTime() / 60000f) + "min"
+                    + ", points0: " + ghResponse.getBest().getPoints().getSize()
+                    + ", debugInfo: " + ghResponse.getDebugInfo());
+            return writeGPX ?
+                    gpxSuccessResponseBuilder(ghResponse, timeString, trackName, enableElevation, withRoute, withTrack, withWayPoints, Constants.VERSION).
+                            header("X-GH-Took", "" + Math.round(took * 1000)).
+                            build()
+                    :
+                    Response.ok(WebHelper.jsonObject(ghResponse, instructions, calcPoints, enableElevation, pointsEncoded, took)).
+                            header("X-GH-Took", "" + Math.round(took * 1000)).
+                            build();
+        }
+    }
+
+    @POST
+    @Consumes({"text/x-yaml", "text/yaml", "application/x-yaml", "application/yaml"})
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, "application/gpx+xml"})
+    public Response doPost(String yaml, @Context HttpServletRequest httpReq) {
+        CustomRequest customRequest;
+        try {
+            customRequest = yamlOM.readValue(yaml, CustomRequest.class);
+        } catch (Exception ex) {
+            // TODO should we really provide this much details to API users?
+            throw new IllegalArgumentException("Incorrect YAML: " + ex.getMessage(), ex);
+        }
+        return doPost(customRequest, httpReq);
+    }
+
+    private static Response.ResponseBuilder gpxSuccessResponseBuilder(GHResponse ghRsp, String timeString, String
+            trackName, boolean enableElevation, boolean withRoute, boolean withTrack, boolean withWayPoints, String version) {
+        if (ghRsp.getAll().size() > 1) {
+            throw new IllegalArgumentException("Alternatives are currently not yet supported for GPX");
+        }
+
+        long time = timeString != null ? Long.parseLong(timeString) : System.currentTimeMillis();
```

Comment:
Here we copy&pasted the bug from #2000 :) I think we should have something like

```java
String createGpx(GHResponse)
```

or at least 

```java
String createGpx(Instructions, PMap)
```

or similar and use it in both `RouteRequest` and `CustomWeightingRouteRequest`. We should also be able to use this in `GET /route` by removing all the parameters like these ` @QueryParam("elevation") @DefaultValue("false") boolean enableElevation` and simply let them be added to the `PMap` such that we can have a single point to specify the defaults. Probably we should fix #2000 and #1996 first.

Reply:
Hmmh, ugly! Do you think we could even remove it completely? (#2006)

Reply:
Ok gpx was removed from this (and the normal /route) endpoint for POST.

Reply:
:+1: